src/tmx/Messages/test/SensorDetectedObjectTest.cpp

```diff
@@ -0,0 +1,111 @@
+        EXPECT_EQ(1.0, tmxSdsmPtr->get_position().x);
+        EXPECT_NEAR(2.3, tmxSdsmPtr->get_position().y, 0.01);
+        EXPECT_EQ(2.0, tmxSdsmPtr->get_position().z);
+        EXPECT_EQ(1.0, tmxSdsmPtr->get_velocity().x);
+        EXPECT_NEAR(0.3, tmxSdsmPtr->get_velocity().y, 0.01);
+        EXPECT_EQ(2.0, tmxSdsmPtr->get_velocity().z);
+        EXPECT_EQ(1.0, tmxSdsmPtr->get_angularVelocity().x);
+        EXPECT_NEAR(0.3, tmxSdsmPtr->get_angularVelocity().y, 0.01);
+        EXPECT_EQ(2.0, tmxSdsmPtr->get_angularVelocity().z);
+        EXPECT_EQ(3, tmxSdsmPtr->get_positionCovariance().size());
+        EXPECT_EQ(3, tmxSdsmPtr->get_angularVelocityCovariance().size());
+        EXPECT_EQ(3, tmxSdsmPtr->get_velocityCovariance().size());
+    }   
+     
+    TEST_F(SensorDetectedObjectTest, to_string){
+        std::string expectedStr = "{\"isSimulated\":\"0\",\"position\":{\"x\":\"1\",\"y\":\"2.2999999999999998\",\"z\":\"2\"},\"projString\":\"+proj=tmerc +lat_0=38.95197911150576 +lon_0=-77.14835128349988 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +geoidgrids=egm96_15.gtx +vunits=m +no_defs\",\"timestamp\":\"12222222222\",\"sensorId\":\"SomeID\",\"type\":\"Car\",\"confidence\":\"0.69999999999999996\",\"objectId\":\"123\",\"velocity\":{\"x\":\"1\",\"y\":\"0.29999999999999999\",\"z\":\"2\"},\"angularVelocity\":{\"x\":\"1\",\"y\":\"0.29999999999999999\",\"z\":\"2\"},\"positionCovariance\":[[\"12\",\"11\",\"13\",\"14\",\"15\"],[\"12\",\"11\",\"13\",\"14\",\"15\"],[\"12\",\"11\",\"13\",\"14\",\"15\"]],\"velocityCovariance\":[[\"12\",\"11\",\"13\",\"14\",\"15\"],[\"12\",\"11\",\"13\",\"14\",\"15\"],[\"12\",\"11\",\"13\",\"14\",\"15\"]],\"angularVelocityCovariance\":[[\"12\",\"11\",\"13\",\"14\",\"15\"],[\"12\",\"11\",\"13\",\"14\",\"15\"],[\"12\",\"11\",\"13\",\"14\",\"15\"]]}\n";
+        EXPECT_EQ(expectedStr, tmxSdsmPtr->to_string());
+    }
+
+    TEST_F(SensorDetectedObjectTest, deserialize){
+        auto tmxSdsmPtr2 = std::make_shared<SensorDetectedObject>();
+        std::string expectedStr = "{\"isSimulated\":1,\"type\":\"CAR\",\"confidence\":1.0,\"sensorId\":\"IntersectionLidar\",\"projString\":\"+proj=tmerc +lat_0=0 +lon_0=0 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +geoidgrids=egm96_15.gtx +vunits=m +no_defs\",\"objectId\":207,\"position\":{\"x\":-5.021,\"y\":64.234,\"z\":-10.297},\"positionCovariance\":[[0.04000000000000001,0.0,0.0],[0.0,0.04000000000000001,0.0],[0.0,0.0,0.04000000000000001]],\"velocity\":{\"x\":0.0,\"y\":0.0,\"z\":0.0},\"velocityCovariance\":[[0.04000000000000001,0.0,0.0],[0.0,0.04000000000000001,0.0],[0.0,0.0,0.04000000000000001]],\"angularVelocity\":{\"x\":0.0,\"y\":-0.0,\"z\":-0.0},\"angularVelocityCovariance\":[[0.010000000000000002,0.0,0.0],[0.0,0.010000000000000002,0.0],[0.0,0.0,0.010000000000000002]],\"size\":{\"length\":2.257,\"height\":1.003,\"width\":0.762},\"timestamp\":110200}";
```

Comment:
**NIT** is there a reason we are not using raw string here. Makes the input more readable.
```cpp
std::string expectedStr = R"(
{
  "isSimulated": 1,
  "type": "CAR",
  "confidence": 1,
  "sensorId": "IntersectionLidar",
  "projString": "+proj=tmerc +lat_0=0 +lon_0=0 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +geoidgrids=egm96_15.gtx +vunits=m +no_defs",
  "objectId": 207,
  "position": {
    "x": -5.021,
    "y": 64.234,
    "z": -10.297
  },
  "positionCovariance": [
    [
      0.04000000000000001,
      0,
      0
    ],
    [
      0,
      0.04000000000000001,
      0
    ],
    [
      0,
      0,
      0.04000000000000001
    ]
  ],
  "velocity": {
    "x": 0,
    "y": 0,
    "z": 0
  },
  "velocityCovariance": [
    [
      0.04000000000000001,
      0,
      0
    ],
    [
      0,
      0.04000000000000001,
      0
    ],
    [
      0,
      0,
      0.04000000000000001
    ]
  ],
  "angularVelocity": {
    "x": 0,
    "y": 0,
    "z": 0
  },
  "angularVelocityCovariance": [
    [
      0.010000000000000002,
      0,
      0
    ],
    [
      0,
      0.010000000000000002,
      0
    ],
    [
      0,
      0,
      0.010000000000000002
    ]
  ],
  "size": {
    "length": 2.257,
    "height": 1.003,
    "width": 0.762
  },
  "timestamp": 110200
}
)"
```